File: frontend/src/services/courtPriceService.js
import axiosInstance from '../axiosConfig/axiosConfig';

const courtPriceService = {
  getAllPrices: async () => {
    const response = await axiosInstance.get('/court-prices');
    return response.data;
  },

  createPrice: async (priceData) => {
    const response = await axiosInstance.post('/court-prices', priceData);
    return response.data;
  },

  updatePrice: async (id, priceData) => {
    const response = await axiosInstance.put(`/court-prices/${id}`, priceData);
    return response.data;
  },

  deletePrice: async (id) => {
    const response = await axiosInstance.delete(`/court-prices/${id}`);
    return response.data;
  },

  getPriceById: async (id) => {
    const response = await axiosInstance.get(`/court-prices/${id}`);
    return response.data;
  },

  getPricesByCourtId: async (courtId) => {
    const response = await axiosInstance.get(`/court-prices/court/${courtId}`);
    return response.data;
  },

  getPricesByCourtIdAndDayType: async (courtId, dayType) => {
    const response = await axiosInstance.get(`/court-prices/court/${courtId}/day-type/${dayType}`);
    return response.data;
  },

  calculatePrice: async (courtId, date, time) => {
    const response = await axiosInstance.get('/court-prices/calculate', {
      params: { courtId, date, time }
    });
    return response.data;
  },

  createDefaultPrices: async (courtId) => {
    const response = await axiosInstance.post(`/court-prices/court/${courtId}/init-default`);
    return response.data;
  }
};

export default courtPriceService;

File: backend/src/main/java/com/badminton/booking/service/CourtPriceService.java
package com.badminton.booking.service;

import com.badminton.booking.dto.request.CourtPriceRequest;
import com.badminton.booking.dto.response.CourtPriceResponse;
import com.badminton.booking.entity.enums.DayType;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalTime;
import java.util.List;

public interface CourtPriceService {

    CourtPriceResponse createPrice(CourtPriceRequest request);

    CourtPriceResponse updatePrice(Integer id, CourtPriceRequest request);

    void deletePrice(Integer id);

    CourtPriceResponse getPriceById(Integer id);

    List<CourtPriceResponse> getPricesByCourtId(Integer courtId);

    List<CourtPriceResponse> getPricesByCourtIdAndDayType(Integer courtId, DayType dayType);

    /**
     * Lấy tất cả giá sân
     */
    List<CourtPriceResponse> getAllPrices();

    /**
     * Lấy giá cho sân tại thời điểm cụ thể
     * 
     * @param courtId ID sân
     * @param date    Ngày (để xác định WEEKDAY/WEEKEND)
     * @param time    Giờ
     * @return Giá/giờ
     */
    BigDecimal getPriceForTime(Integer courtId, LocalDate date, LocalTime time);

    /**
     * Tạo bảng giá mặc định cho sân
     */
    List<CourtPriceResponse> createDefaultPricesForCourt(Integer courtId);

    /**
     * Tạo nhiều giá cùng lúc (batch), tùy chọn xóa giá cũ
     */
    java.util.Map<String, Object> createPricesBatch(List<CourtPriceRequest> requests, boolean deleteExisting);
}
